containers/src/Kokkos_ScatterView.hpp

```diff
@@ -57,9 +57,16 @@
 namespace Kokkos {
 namespace Experimental {
 
-//TODO: replace this enum with the Kokkos::Sum, etc reducers for parallel_reduce
+/*
+ * Reduction Type list
+ *  - These corresponds to subset of the reducers in parallel_reduce
+ *  - See Implementations of ScatterValue for details.
+ */
 enum : int {
   ScatterSum,
+  ScatterProd,
```

Comment:
It's a bit annoying that Kokkos has separate reducer objects (e.g., Sum, Min, Max) and scatter reducer enums.  Would it make sense to consolidate the enums, and have `parallel_reduce` use the enum to point to a built-in reducer object, just like how e.g., `MPI_MAX` points to a built-in reduction function?

Reply:
There has been talk of this, and I think a central `enum` makes sense. @crtrott would like to use the existing reducer classes (e.g. `Sum`). I think its beyond the scope of this PR, and would break compatibility.